ui/components/app/modals/qr-scanner/qr-scanner.component.js

```diff
@@ -30,8 +30,8 @@ const parseContent = (content) => {
   // Ethereum address links - fox ex. ethereum:0x.....1111
   if (content.split('ethereum:').length > 1) {
     type = 'address';
-    values = { address: content.split('ethereum:')[1] };
-
+    // uses regex capture groups to match and extract address while ignoring everythign else
+    values = { address: content.match(/.*:(0x[0-9a-fA-F]{40})(?:@.*)?/)[1] }
```

Comment:
I hate pointing this out but should we have this regex in a function so we can add a test?  Also, is there a product consideration here?  i.e. if the string ends in `0x89`, and the user is on Mainnet, should something happen?

Reply:
+1 on test

> . if the string ends in 0x89, and the user is on Mainnet, should something happen?

perhaps though I feel that's outside the scope of the bug.